### ISSUE TITLE
Ban/kick entry in user context menu

### ISSUE THREAD

**Issue by YaManicKill** [thelounge/lounge#347]
Would be nice for ops to have a way to ban, kick or kickban a user from the context menu.

**Comment by astorije** [comment]
This was addressed by #1722. @YaManicKill, can we close this issue? I know not all actions have been added, so I'll leave it up to you. It would be nice to link this to 2.7 and call it a go :D

**Comment by MaxLeiter** [comment]
I think this should be left open, as ban is a pretty fundamental and expected feature. Perhaps we could make a new issue for ban? However the larger issue with that is having to refactor to get a users host name and such

**Comment by YaManicKill** [comment]
I'd be happy to close this one if we create a new one for adding ban to the context menu. But yeah, I agree with @MaxLeiter , I think the ban functionality is pretty big and we should have an open issue for it.

**Comment by astorije** [comment]
Fair enough, I was just checking in on a couple old issues. Ok with whatever you prefer.

**Comment by YaManicKill** [comment]
Let's just keep this in, but I'll edit it, then. It's not like we have a counter for "number of closed issues" just "number of opened issues".

**Comment by plusin** [comment]
Please add ban option! its fundamental to the operators of rooms.

**Comment by richrd** [comment]
So, what exactly should be banned when banning a user via the context menu?

**Comment by Mikaela** [comment]
I like WeeChat's default which is banning `*!$ident@$host` meaning that if there is no identd `*!*@host` willl get baned and if there is identd, `*!ident@host` (e.g. `*!uid123456@irccloud.com` will get banned.

However it won't work on networks that have no identd checking (and thus no ~ in ident) or if the host allows users to change the ident at will anytime, so maybe ban should be a submenu offering banning options such as `*!$ident@$host`, `*!*@$host`, I don't know how useful `*nick*!ident@host or variations of that would be as nick is the easiest one to change.
